bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-scripts.js

```diff
@@ -283,11 +283,11 @@ export default function defineOHBlocks_Scripts (f7, isGraalJs, scripts) {
     if (contextInfo === 'ruleUID') return ['ctx.ruleUID', javascriptGenerator.ORDER_ATOMIC]
     if (contextInfo === 'itemState' || contextInfo === 'oldItemState' || contextInfo === 'itemCommand') {
       if (type === 'asNumber') {
-        return [`parseFloat(event.${contextInfo}.toString())`, javascriptGenerator.ORDER_ATOMIC]
+        return [`parseFloat(event.${contextInfo}?.toString())`, javascriptGenerator.ORDER_ATOMIC]
```

Comment:
Optional chaining inside another method doesn't work:

```suggestion
        return [`event.${contextInfo} !== undefined ? parseFloat(event.${contextInfo}.toString()) : undefined`, javascriptGenerator.ORDER_ATOMIC]
```

parseFloat will return NaN if it gets passed in undefined, which will then fail the undefined check in a if-statement created by the user.